pkg/alertmanager/api.go

```diff
@@ -496,6 +507,20 @@ func validateGlobalConfig(cfg config.GlobalConfig) error {
 	return nil
 }
 
+// validateDiscordConfig validates the Discord config and returns an error if it
+// contains settings not allowed by Mimir.
+func validateDiscordConfig(cfg config.DiscordConfig) error {
+	if cfg.WebhookURLFile != "" {
+		return errWebhookURLFileNotAllowed
+	}
+	if cfg.HTTPConfig != nil {
```

Comment:
I don't think this is required because the validation is performed recursively, the method only needs to validate non-struct fields on this config. Can you confirm with a test?

Reply:
You are correct. I removed this code, but the tests I added `*DiscordConfig.HTTPConfig`, `DiscordConfig.HTTPConfig`, `*MSTeams.HTTPConfig` and `MSTeams.HTTPConfig` still pas.

Reply:
You are correct. I removed this code, and the tests I added *DiscordConfig.HTTPConfig, DiscordConfig.HTTPConfig, *MSTeams.HTTPConfig and MSTeams.HTTPConfig still pass.